clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -338,7 +339,8 @@ public void onGroupAssignmentUpdated(Set<TopicPartition> partitions) {
 
             ApiVersions apiVersions = new ApiVersions();
             final BlockingQueue<ApplicationEvent> applicationEventQueue = new LinkedBlockingQueue<>();
-            final BackgroundEventHandler backgroundEventHandler = new BackgroundEventHandler(backgroundEventQueue);
+            this.backgroundEventHandler = new BackgroundEventHandler(
+                backgroundEventQueue, time, asyncConsumerMetrics);
```

Comment:
Nitpick: for better or worse, we've adopted this style for multi-line parameter lists:

```suggestion
            this.backgroundEventHandler = new BackgroundEventHandler(
                backgroundEventQueue,
                time,
                asyncConsumerMetrics
            );
```